builder/dockerfile/parser/parser.go

```diff
@@ -176,17 +176,19 @@ func Parse(rwc io.Reader, d *Directive) (*Node, error) {
 				newline := scanner.Text()
 				currentLine++
 
-				// If escape followed by a comment line then stop
+				// If escape followed by an empty line then stop
+				if strings.TrimSpace(newline) == "" {
+					break
+				}
+
+				// If escape followed by a comment line then we
+				// continue.
 				// Note here that comment line starts with `#` at
 				// the first pos of the line
 				if stripComments(newline) == "" {
```

Comment:
This should be modified to match the "old" matching of comments. Due to the "weird" parsing of empty lines in the old behavior, comments _not_ starting on position 0 were actually seen as empty lines / comments.

`stripComments()` only matches comments that start on position 0, so something like;

```golang
if strings.TrimSpace(newline)[0] == '#' {
    continue
}
```